tsdb/blockwriter_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tsdb
+
+import (
+	"context"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/tsdb/tsdbutil"
+	"github.com/prometheus/prometheus/util/testutil"
+)
+
+func TestBlockWriter(t *testing.T) {
+	ctx := context.Background()
+	outputDir, err := ioutil.TempDir(os.TempDir(), "output")
+	testutil.Ok(t, err)
+	w, err := NewBlockWriter(log.NewNopLogger(), outputDir, DefaultBlockDuration)
+	testutil.Ok(t, err)
+
+	// Flush with no series results in error.
+	_, err = w.Flush(ctx)
+	testutil.ErrorEqual(t, err, errors.New("no series appended, aborting"))
+
+	// Add some series.
+	app := w.Appender(ctx)
+	expectedLabelNames := []string{"a", "c"}
+	expectedValues := []string{"b", "d"}
+	l := labels.Labels{{Name: expectedLabelNames[0], Value: expectedValues[0]}}
+	ts1 := int64(44)
+	v1 := float64(7)
+	_, err = app.Add(l, ts1, v1)
+	testutil.Ok(t, err)
+	l = labels.Labels{{Name: expectedLabelNames[1], Value: expectedValues[1]}}
+	ts2 := int64(55)
+	v2 := float64(12)
+	_, err = app.Add(l, ts2, v2)
+	testutil.Ok(t, err)
+	err = app.Commit()
+	testutil.Ok(t, err)
+	id, err := w.Flush(ctx)
+	testutil.Ok(t, err)
```

Comment:
We can make this a little more compact, else it's little harder to read
```suggestion
	ts1, v1 := int64(44), float64(7)
	_, err = app.Add(labels.Labels{{Name: "a", Value: "b"}}, ts1, v1)
	testutil.Ok(t, err)
	ts2, v2 := int64(55), float64(12)
	_, err = app.Add(labels.Labels{{Name: "c", Value: "d"}}, ts2, v2)
	testutil.Ok(t, err)
	testutil.Ok(t, app.Commit())
	id, err := w.Flush(ctx)
	testutil.Ok(t, err)
```